src/state/haproxy_route.py

```diff
@@ -254,10 +286,27 @@ def get_servers_definition_from_requirer_data(
             servers.append(
                 HAProxyRouteServer(
                     server_name=f"{requirer.application_data.service}_{port}_{i}",
-                    host=unit_data.host,
+                    address=unit_data.address,
                     port=port,
                     check=requirer.application_data.check,
                     maxconn=requirer.application_data.server_maxconn,
                 )
             )
     return servers
+
+
+def get_backend_max_path_depth(backend: HAProxyRouteBackend) -> int:
+    """Return the max depth of requested paths for the given backend.
+
+    Return 1 if no custom path is requested.
+
+    Args:
+        backend: haproxy-route backend.
+
+    Returns:
+        int: The max requested path depth
+    """
+    paths = backend.application_data.paths
+    if not paths:
+        return 1
+    return max(len(path) for path in paths)
```

Comment:
You need to split on "/" to get the depth

Reply:
added, thanks!